scripts/aptly/aptly_importer.py

```diff
@@ -0,0 +1,338 @@
+import argparse
+from collections import defaultdict
+from enum import Enum
+from pathlib import Path
+from os import path
+from subprocess import check_output, PIPE, run, CalledProcessError
+from sys import stderr, exit
+import re
+import time
+import yaml
+
+
+class Reprepro2AptlyFilter():
+    def convert(self, filter_str):
+        # remove begin/end whitespaces
+        r_str = filter_str.strip()
+        # Package is not a valid Aptly filter. Could be Name or empty.
+        r_str = r_str.replace('Package', 'Name')
+        # Aptly filters use (= value) for exact matches rather than (% =value)
+        r_str = r_str.replace('% =', '= ')
+        return r_str
+
+
+class Aptly():
+    class ArtifactType(Enum):
+            MIRROR = 'mirror'
+            REPOSITORY = 'repo'
+            SNAPSHOT = 'snapshot'
+            PUBLISH = 'publish'
+
+    def __init__(self, debug=False, config_file=None):
+        self.debug = debug
+        self.config_file = config_file
+
+    def __error(self, cmd, msg, exit_on_errors=False):
+        print(f"Aptly error running: {cmd}", file=stderr)
+        print(f"  --> {msg} \n", file=stderr)
+        if exit_on_errors:
+            exit(-1)
```

Comment:
Does it make sense to use a low valued exit code rather than -1/255?

Reply:
Used 1 for generic errors, 2 for config file errors. 7e74b46